picard/ui/ui_options_general.py

```diff
@@ -67,58 +67,61 @@ def setupUi(self, GeneralOptionsPage):
-        self.groupBox_3.setObjectName("groupBox_3")
-        self.verticalLayout_2 = QtWidgets.QVBoxLayout(self.groupBox_3)
-        self.verticalLayout_2.setObjectName("verticalLayout_2")
-        self.check_for_updates = QtWidgets.QCheckBox(self.groupBox_3)
-        self.check_for_updates.setObjectName("check_for_updates")
-        self.verticalLayout_2.addWidget(self.check_for_updates)
-        self.gridLayout = QtWidgets.QGridLayout()
-        self.gridLayout.setContentsMargins(-1, -1, -1, 0)
-        self.gridLayout.setObjectName("gridLayout")
-        self.label_2 = QtWidgets.QLabel(self.groupBox_3)
-        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.MinimumExpanding, QtWidgets.QSizePolicy.Preferred)
-        sizePolicy.setHorizontalStretch(0)
-        sizePolicy.setVerticalStretch(0)
-        sizePolicy.setHeightForWidth(self.label_2.sizePolicy().hasHeightForWidth())
-        self.label_2.setSizePolicy(sizePolicy)
-        self.label_2.setObjectName("label_2")
-        self.gridLayout.addWidget(self.label_2, 0, 0, 1, 1)
-        self.update_check_days = QtWidgets.QSpinBox(self.groupBox_3)
-        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Fixed, QtWidgets.QSizePolicy.Fixed)
-        sizePolicy.setHorizontalStretch(0)
-        sizePolicy.setVerticalStretch(0)
-        sizePolicy.setHeightForWidth(self.update_check_days.sizePolicy().hasHeightForWidth())
-        self.update_check_days.setSizePolicy(sizePolicy)
-        self.update_check_days.setAlignment(QtCore.Qt.AlignLeading|QtCore.Qt.AlignLeft|QtCore.Qt.AlignVCenter)
-        self.update_check_days.setMinimum(1)
-        self.update_check_days.setObjectName("update_check_days")
-        self.gridLayout.addWidget(self.update_check_days, 0, 1, 1, 1)
-        self.verticalLayout_2.addLayout(self.gridLayout)
-        self.gridLayout_2 = QtWidgets.QGridLayout()
-        self.gridLayout_2.setContentsMargins(-1, -1, -1, 0)
-        self.gridLayout_2.setObjectName("gridLayout_2")
-        self.label_3 = QtWidgets.QLabel(self.groupBox_3)
-        self.label_3.setObjectName("label_3")
-        self.gridLayout_2.addWidget(self.label_3, 0, 0, 1, 1)
-        self.update_level = QtWidgets.QComboBox(self.groupBox_3)
-        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.MinimumExpanding, QtWidgets.QSizePolicy.Fixed)
-        sizePolicy.setHorizontalStretch(0)
-        sizePolicy.setVerticalStretch(0)
-        sizePolicy.setHeightForWidth(self.update_level.sizePolicy().hasHeightForWidth())
-        self.update_level.setSizePolicy(sizePolicy)
-        self.update_level.setEditable(False)
-        self.update_level.setObjectName("update_level")
-        self.gridLayout_2.addWidget(self.update_level, 0, 1, 1, 1)
-        self.verticalLayout_2.addLayout(self.gridLayout_2)
-        self.vboxlayout.addWidget(self.groupBox_3)
+
+        if QtWidgets.QApplication.instance().autoupdate_enabled:
+            self.groupBox_3 = QtWidgets.QGroupBox(GeneralOptionsPage)
```

Comment:
Since we're likely to be referring to this to hide it, the name should probably be changed from `groupbox_3` to something more appropriate (e.g. `update_check_groupbox`).